tests/integration/test_transcribe.py

```diff
@@ -180,3 +202,73 @@ def test_failing_start_transcription_job(self, s3_bucket, snapshot, aws_client):
+    )
+    @pytest.mark.parametrize(
+        "output_bucket,output_key",
+        [
+            ("test-output-bucket-2", None),  # with output bucket and without output key
+            (
+                "test-output-bucket-3",
+                "test-output",
+            ),  # with output bucket and output key without .json
+            (
+                "test-output-bucket-4",
+                "test-output.json",
+            ),  # with output bucket and output key with .json
+            (
+                "test-output-bucket-5",
+                "test-files/test-output.json",
+            ),  # with output bucket and with folder key with .json
+            (
+                "test-output-bucket-6",
+                "test-files/test-output",
+            ),  # with output bucket and with folder key without .json
+            (None, None),  # without output bucket and output key
+        ],
+    )
+    def test_transcribe_start_job(
+        self, output_bucket, output_key, s3_bucket, cleanups, snapshot, aws_client
+    ):
+        file_path = os.path.join(BASEDIR, "files/en-gb.wav")
+        test_key = "test-clip.wav"
+        transcribe_job_name = f"test-transcribe-job-{short_uid()}"
+        params = {
+            "TranscriptionJobName": transcribe_job_name,
+            "LanguageCode": "en-GB",
+            "Media": {"MediaFileUri": f"s3://{s3_bucket}/{test_key}"},
+        }
+
+        if output_bucket is not None:
+            params["OutputBucketName"] = output_bucket
+            aws_client.s3.create_bucket(Bucket=output_bucket)
+        if output_key is not None:
+            params["OutputKey"] = output_key
+
+        with open(file_path, "rb") as f:
+            aws_client.s3.upload_fileobj(f, s3_bucket, test_key)
+
+        response_start_job = aws_client.transcribe.start_transcription_job(**params)
+        self._wait_transcription_job(aws_client.transcribe, params["TranscriptionJobName"])
+        snapshot.match("response-start-job", response_start_job)
+        response_get_transcribe_job = aws_client.transcribe.get_transcription_job(
+            TranscriptionJobName=transcribe_job_name
+        )
+        snapshot.match("response-get-transcribe-job", response_get_transcribe_job)
+
+        res_delete_transcription_job = aws_client.transcribe.delete_transcription_job(
+            TranscriptionJobName=transcribe_job_name
+        )
+        snapshot.match("delete-transcription-job", res_delete_transcription_job)
+
+        if output_bucket:
+            objects = aws_client.s3.list_objects_v2(Bucket=output_bucket)
+            if "Contents" in objects:
+                for obj in objects["Contents"]:
+                    cleanups.append(
+                        lambda: aws_client.s3.delete_object(Bucket=output_bucket, Key=obj["Key"])
+                    )
+            cleanups.append(lambda: aws_client.s3.delete_bucket(Bucket=output_bucket))
```

Comment:
This is incorrect. This still won't run if any of the earlier assertions fail. cleanup appends must happen immediately after the resource is created, so on line 247 immediately after the bucket is created.

Create a function with cleanup logic and use it with the fixture.

```py
def _cleanup():
  objects = aws_client.s3.list_objects_v2(Bucket=output_bucket)
  if "Contents" in objects:
    for obj in objects["Contents"]:
      aws_client.s3.delete_object(Bucket=output_bucket, Key=obj["Key"])
  aws_client.s3.delete_bucket(Bucket=output_bucket)

cleanups.append(_cleanup)
```

You can test it by `assert False` and making sure the resources are removed.